GitUI/CommandsDialogs/FormStash.Designer.cs

```diff
@@ -120,34 +120,46 @@ private void InitializeComponent()
-            this.tableLayoutPanel1.Controls.Add(this.Apply, 0, 3);
-            this.tableLayoutPanel1.Controls.Add(this.Clear, 0, 2);
+            this.tableLayoutPanel1.Controls.Add(this.Apply, 0, 4);
+            this.tableLayoutPanel1.Controls.Add(this.Clear, 0, 3);
             this.tableLayoutPanel1.Dock = System.Windows.Forms.DockStyle.Fill;
-            this.tableLayoutPanel1.Location = new System.Drawing.Point(4, 494);
-            this.tableLayoutPanel1.Margin = new System.Windows.Forms.Padding(4);
+            this.tableLayoutPanel1.Location = new System.Drawing.Point(3, 370);
             this.tableLayoutPanel1.Name = "tableLayoutPanel1";
             this.tableLayoutPanel1.RowCount = 5;
             this.tableLayoutPanel1.RowStyles.Add(new System.Windows.Forms.RowStyle());
             this.tableLayoutPanel1.RowStyles.Add(new System.Windows.Forms.RowStyle());
             this.tableLayoutPanel1.RowStyles.Add(new System.Windows.Forms.RowStyle());
             this.tableLayoutPanel1.RowStyles.Add(new System.Windows.Forms.RowStyle());
             this.tableLayoutPanel1.RowStyles.Add(new System.Windows.Forms.RowStyle());
-            this.tableLayoutPanel1.Size = new System.Drawing.Size(272, 152);
+            this.tableLayoutPanel1.Size = new System.Drawing.Size(274, 147);
             this.tableLayoutPanel1.TabIndex = 5;
             // 
+            // StashSelectedFiles
+            // 
+            this.tableLayoutPanel1.SetColumnSpan(this.StashSelectedFiles, 2);
+            this.StashSelectedFiles.Dock = System.Windows.Forms.DockStyle.Fill;
+            this.StashSelectedFiles.Location = new System.Drawing.Point(3, 57);
+            this.StashSelectedFiles.Name = "StashSelectedFiles";
+            this.StashSelectedFiles.Size = new System.Drawing.Size(268, 25);
+            this.StashSelectedFiles.TabIndex = 18;
+            this.StashSelectedFiles.Text = "Stash selected changes";
+            this.toolTip.SetToolTip(this.StashSelectedFiles, "Save local changes of selected files to a new stash, then revert local changes");
+            this.StashSelectedFiles.UseVisualStyleBackColor = true;
+            this.StashSelectedFiles.Click += new System.EventHandler(this.StashSelectedFiles_Click);
+            // 
             // Stash
             // 
             this.tableLayoutPanel1.SetColumnSpan(this.Stash, 2);
             this.Stash.Dock = System.Windows.Forms.DockStyle.Fill;
-            this.Stash.Location = new System.Drawing.Point(4, 39);
-            this.Stash.Margin = new System.Windows.Forms.Padding(4);
+            this.Stash.Location = new System.Drawing.Point(3, 26);
             this.Stash.Name = "Stash";
-            this.Stash.Size = new System.Drawing.Size(264, 31);
+            this.Stash.Size = new System.Drawing.Size(268, 25);
             this.Stash.TabIndex = 15;
-            this.Stash.Text = "Save Changes to New Stash";
+            this.Stash.Text = "Stash changes";
```

Comment:
Given that a user can stash _selected_ changes, perhaps it is better to say _Stash all changes_